src/librustc/dep_graph/dep_node.rs

```diff
@@ -570,8 +570,8 @@ define_dep_nodes!( <'tcx>
     [] UsedCrateSource(CrateNum),
     [] PostorderCnums,
 
-    [] Freevars(HirId),
-    [] MaybeUnusedTraitImport(HirId),
+    [] Freevars(DefIndex),
```

Comment:
Why not use `DefId` here? Closures have `DefId`s.

Reply:
Indeed! Got a bit overzealous with `DefIndex`